platform/ios/src/MGLMapView.mm

```diff
@@ -1566,7 +1604,11 @@ - (void)handleDoubleTapGesture:(UITapGestureRecognizer *)doubleTap
             [self animateWithDelay:MGLAnimationDuration animations:^
              {
                  [weakSelf unrotateIfNeededForGesture];
+                 [weakSelf notifyScaleBarGestureDidEnd];
              }];
+        } else {
+            [self unrotateIfNeededForGesture];
+            [self notifyScaleBarGestureDidEnd];
```

Comment:
@frederoni I think we have to keep the scale bar unhiding in `-cameraIsChanging`, otherwise programmatic zooming won’t trigger it. If that’s true, then bifurcating `-notifyGestureDidBegin` wouldn’t do us any good. (Have we considered programmatic zooming?)

If we don’t care about programmatic zoom triggering the scale bar, then perhaps we can keep a single method but have it take an enum, e.g., `-notifyGestureDidBeginWithType:` and `MGLGestureType`, that would define whether or not it was initiated by a zoom-related gesture. (Or just a `-notifyGestureDidBeginWithZoom:` boolean.)